include/ppx/string_util.h

```diff
@@ -31,9 +37,17 @@ std::string TrimCopy(const std::string& s);
 // Trims all characters specified in c from both the left and right sides of s
 std::string_view TrimBothEnds(std::string_view s, std::string_view c = " \t");
 
+// Splits s at every instance of delimeter and returns a vector of substrings
+// Return value will contain empty substrings if s has leading/trailing/consecutive delimiters
+std::vector<std::string_view> Split(std::string_view s, char delimiter);
+
 // Splits s at the first instance of delimeter and returns two substrings
-// Returns std::nullopt if s does not contain the delimeter
-std::optional<std::pair<std::string_view, std::string_view>> SplitInTwo(std::string_view s, char delimiter);
+// Returns an empty pair if s does not have exactly one delimiter, or if either of the substrings are empty
```

Comment:
Since this is a string utility that can be used for other things, I would actually go one step further, and split on the first instance of delimeter, and return everything pre and everything post without making any assumptions.

If there is no delimiter, the second part will be empty, if there is more than one delimeter the second value will have those delimeters, if the strings are empty, return empty strings, then allow the caller to decide what to do with those return values.
With the current implementation it is acceptable to pass:
`--list-gpus ""`
but the following would produce an error: 
`--list-gpus=""`
which sounds inconsistent

Reply:
I agree, made those changes.
I changed a bit of `Commandline::Parse` logic so that empty value is allowed and empty option name is caught by other checks, added appropriate unit tests.